distribution/docs/src/main/resources/_contents/_endpoints/cometd-endpoint-contents.adoc

```diff
@@ -0,0 +1,855 @@
+
+.CometD Activity Format
+[cols="1m,3,1,1,1" options="header"]
+|===
+|Map Key
+|Description
+|Value Type
+|Possible Values
+|Example Values
+
+|category
+|Category of the activity
+|String
+|Any String
+|"Product Retrieval"
+
+|id
+|ID that uniquely identifies the activity that sent out the update. Not required
+to be unique per update.
+|String
+|Any String
+|"b72ccdf6-8ca7-4f53-a0f6-b0ad264393b0"
+
+|message
+|User-readable message that explains the current activity status
+|String
+|Any String
+|"Resource retrieval downloading."
+
+|operations
+|Map of operations that can be performed on this activity
+|JSON Map
+|A map of keys with populated values (that evaluate to 'true' rather than 'null' 'undefined' or 'false') These operations and their values can be used by clients to communicate back to the server by sending a message back on the same channel.
+
+If the value is a URL, the client should invoke the URL as a result of the user invoking the activity operation.
+
+a|
+[source,json,linenums]
+----
+"operations" : {
+	"download" : "http://example.com/product"
+}
+----
+
+If the value is not a URL, the client should send a message back to the server on the same topic with the operation name.
+
+Note: the ${ddf-branding} UI will interpret several values with special icons:
+* `download`
+* `retry`
+* `cancel`
+* `pause`
+* `remove`
+* `resume`
+
+|progress
+|Percentage value of activity completion
+|String
+|Integer between 0 - 100 followed by a %
+|"45%"
+
+|status
+|Enumerated value that displays the current state of the activity
+|String
+a|
+* `STARTED`,
+* `RUNNING`,
+* `FINISHED`,
+* `STOPPED`,
+* `PAUSED`, or
+* `FAILED`
+|`RUNNING`
+
+|timestamp
+|Time that the activity update was sent
+|String
+|Positive long value (seconds
+since unix epoch)
+|`1403801920875`
+
+|title
+|User-readable title for the activity update
+|String
+|Any String
+|"Download Complete"
+
+|subject
+|User who started the activity
+|String
+|Any String
+|"admin"
+
+|bytes
+|Number of bytes the activity consumed (upload or download)
+|Integer
+|Any Integer
+|11
+
+|session
+|The session ID of the user/subject
+|String
+|Any String
+|"6q2lfmmwh9dwv1c4gevboyp297"
+
+|Custom Value
+|Additional keys can be inserted by the component sending the activity notification
+|Any JSON Type
+|Any JSON
+|"{}"
+
+|===
+
+.Example: Activity update with custom 'bytes' field
+[source,json,linenums]
+----
+data: {
+ 	"category": "Product Retrieval",
+ 	"event.topics": "ddf/activities/broadcast",
+ 	"id": "a62f6666-fc41-4a19-91f1-485e73a564b5",
+ 	"message": "The requested product is being retrieved.
+		Standby.",
+ 	"operations": {
+ 		"cancel" : true
+	},
+ 	"progress": "",
+	"status": "RUNNING",
+	"timestamp": "1403801920875",
+ 	"title": "Product downloading",
+ 	"user": "admin",
+ 	"bytes": 635084800
+}
+----
+
+====== Activity Operations Channel
+
+An activity operation is published to the channel `/service/action`
```

Comment:
Can we improve this description? Something about the fact that different operations can be performed on activities using the `/service/action` channel.